orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/metrics/AtlasQueueMonitor.kt

```diff
@@ -71,6 +84,23 @@ class AtlasQueueMonitor
     _lastState.set(queue.readState())
   }
 
+  @Scheduled(fixedDelayString = "\${queue.zombie.check.frequency:3600000}")
+  fun checkForZombies() {
+    val startedAt = clock.instant()
+    repository.retrieve(PIPELINE)
+      .mergeWith(repository.retrieve(ORCHESTRATION))
+      .subscribeOn(zombieCheckScheduler.orElseGet(Schedulers::io))
+      .filter { it.status == RUNNING }
+      .filter(this::hasBeenAroundAWhile)
+      .filter(this::queueHasNoMessages)
+      .doOnCompleted {
+        log.info("Completed zombie check in ${Duration.between(startedAt, clock.instant())}")
+      }
+      .subscribe {
+        registry.counter("zombie.execution", "id", it.id).increment()
```

Comment:
We definitely don't want do add the execution ID as a tag. Would much prefer a simple counter, and then log more specific information.

Reply:
It would be nice if there was enough info in the alert to at least point you at something useful. Application? Pipeline name?

Reply:
Well, no. Orchestrations do have a name though, but if you're worried about tag bloat it's probably not a good idea to use that.